### PR TITLE
Refactor preprocessors

### PR DESCRIPTION
**Pull request by Mupico**
issue #48 , refactoring recalculation tokenise text such that it can be used by both text metadata and text embedding.

### CODE REVIEW

File: shift_detector/precalculations/Tokenizer.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import re
+import pandas as pd
+
+from shift_detector.precalculations.Precalculation import Precalculation
+from shift_detector.utils.ColumnManagement import ColumnType
+
+class TokenizeIntoWords(Precalculation):
```
Comment (janwestphal306) [inline shift_detector/precalculations/Tokenizer.py]: The class name should also reflect that the text is converted to lowercase and should end with `~Precalculation`. Like `LowerTokenizePrecalculation`

File: shift_detector/precalculations/TextEmbeddingPrecalculation.py [hunk 2]
```diff
@@ -49,20 +50,17 @@ def __hash__(self):
...
         # return vector of zeros when cell was empty
         if type(ser) == np.float64:
             ser = [0.0] * self.trained_model.vector_size
         return ser
 
     def process(self, store: Store):
-        df1, df2 = store[ColumnType.text]
-        df1 = df1.copy().dropna()
-        df2 = df2.copy().dropna()
+        df1, df2 = store[TokenizeIntoWords()]
```
Comment (janwestphal306) [inline shift_detector/precalculations/TextEmbeddingPrecalculation.py]: This variable names should be more descriptive. Like `df1_lower_tokenized`

File: tests/precalculations/test_text_preprocessor.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import unittest
+
+from shift_detector.precalculations.Tokenizer import TokenizeIntoWords
+
+class TestTokenizerFunctions(unittest.TestCase):
+
+    def test_tokenize_into_words(self):
+        normal = "This. is a'n example, ,, 12  35,6  , st/r--ing    \n test."
```
Comment (janwestphal306) [inline tests/precalculations/test_text_preprocessor.py]: Is it desired behavior to ignore all commas in this string?

Comment (Mupico) [reply tests/precalculations/test_text_preprocessor.py]: yes

File: shift_detector/precalculations/TextEmbeddingPrecalculation.py [hunk 4]
```diff
@@ -49,30 +50,27 @@ def __hash__(self):
...
 
     def process(self, store: Store):
-        df1, df2 = store[ColumnType.text]
-        df1 = df1.copy().dropna()
-        df2 = df2.copy().dropna()
+        df1_tokenized, df2_tokenized = store[TokenizeIntoLowerWordsPrecalculation()]
 
-        concatenated_ser = pd.concat([df1[i].str.lower().str.split() for i in df1] +
-                                     [df2[i].str.lower().str.split() for i in df2])
+        concatenated_ser = pd.concat([df1_tokenized[i] for i in df1_tokenized] + [df2_tokenized[i] for i in df2_tokenized])
```
Comment (phSch08) [inline shift_detector/precalculations/TextEmbeddingPrecalculation.py]: Line is to long here -> add linebreak

File: shift_detector/precalculations/TextEmbeddingPrecalculation.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
 from numbers import Number
 from copy import copy
 from shift_detector.precalculations.Precalculation import Precalculation
+from shift_detector.precalculations.Tokenizer import TokenizeIntoLowerWordsPrecalculation
 from shift_detector.precalculations.Store import Store, ColumnType
```
Comment (phSch08) [inline shift_detector/precalculations/TextEmbeddingPrecalculation.py]:
```suggestion
from shift_detector.precalculations.Store import Store
```

File: shift_detector/precalculations/TextMetadata.py [hunk 6]
```diff
@@ -319,7 +292,7 @@ def metadata_function(self, language, words):
             if len(words) == 0:
                 return 0.0
             for word in words:
-                if word.lower() in stop:
+                if word in stop:
```
Comment (phSch08) [inline shift_detector/precalculations/TextMetadata.py]: I know you didn't touch this, but 'stop' is very undescriptive. Can you replace it with a more descriptive variable name?

File: shift_detector/precalculations/Tokenizer.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import re
```
Comment (phSch08) [inline shift_detector/precalculations/Tokenizer.py]: Please rename file to something with 'Preprocessor'

File: shift_detector/precalculations/Tokenizer.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+import pandas as pd
+
+from shift_detector.precalculations.Precalculation import Precalculation
+from shift_detector.utils.ColumnManagement import ColumnType
+
```
Comment (phSch08) [inline shift_detector/precalculations/Tokenizer.py]:
```suggestion


```

Add one line here

File: shift_detector/precalculations/Tokenizer.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+class TokenizeIntoLowerWordsPrecalculation(Precalculation):
+
+    def __eq__(self, other):
+        return isinstance(other, self.__class__)
+
+    def __hash__(self):
+        return hash(self.__class__)
+
+    def tokenize_into_words(self, text):
```
Comment (phSch08) [inline shift_detector/precalculations/Tokenizer.py]:
```suggestion
    @staticmethod
    def tokenize_into_words(self, text):
```

File: tests/precalculations/test_text_preprocessor.py [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+from shift_detector.precalculations.Store import Store
+from shift_detector.precalculations.Tokenizer import *
+
+class TestTokenizerFunctions(unittest.TestCase):
+
+    def setUp(self):
+        poems = [
+            'Tell me not, in mournful numbers,\nLife is but an empty dream!\nFor the soul is dead that slumbers,\nAnd things are not what they seem.',
+            'Life is real! Life is earnest!\nAnd the grave is not its goal;\nDust thou art, to dust returnest,\nWas not spoken of the soul.'
```
Comment (phSch08) [inline tests/precalculations/test_text_preprocessor.py]: Lines 12 and 13 are to long

File: tests/precalculations/test_text_preprocessor.py [hunk 11]
```diff
@@ -0,0 +1,41 @@
+import unittest
+
+from pandas.util.testing import assert_frame_equal
+
+from shift_detector.precalculations.Store import Store
+from shift_detector.precalculations.Tokenizer import *
+
```
Comment (phSch08) [inline tests/precalculations/test_text_preprocessor.py]:
```suggestion


```

two free lines before class

File: tests/precalculations/test_text_preprocessor.py [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+            'Front-line leading edge website',
+            'Upgradable upward-trending software'
+        ]
+        df1 = pd.DataFrame.from_dict({'text': poems})
+        df2 = pd.DataFrame.from_dict({'text': phrases})
+        self.store = Store(df1, df2)
+
+    def test_tokenize_into_words_preprocessor(self):
+        md1, md2 = TokenizeIntoLowerWordsPrecalculation().process(self.store)
+        sol1_1 = ['tell', 'me', 'not', 'in', 'mournful', 'numbers', 'life', 'is', 'but', 'an', 'empty', 'dream', 'for', 'the', 'soul', 'is', 'dead', 'that', 'slumbers', 'and', 'things', 'are', 'not', 'what', 'they', 'seem']
```
Comment (phSch08) [inline tests/precalculations/test_text_preprocessor.py]: line to long

File: tests/precalculations/test_text_preprocessor.py [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+        sol1_2 = ['life', 'is', 'real', 'life', 'is', 'earnest', 'and', 'the', 'grave', 'is', 'not', 'its', 'goal', 'dust', 'thou', 'art', 'to', 'dust', 'returnest', 'was', 'not', 'spoken', 'of', 'the', 'soul']
```
Comment (phSch08) [inline tests/precalculations/test_text_preprocessor.py]: line to long

File: tests/precalculations/test_text_preprocessor.py [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+        solution2 = pd.DataFrame([[sol2_1], [sol2_2]], columns=['text'])
+        assert_frame_equal(solution1, md1)
+        assert_frame_equal(solution2, md2)
+
+    def test_tokenize_into_words_function(self):
+        normal = "This. is a'n example, ,, 12  35,6  , st/r--ing    \n test."
+        empty = ""
+        punctuation = ".  , * (  \n \t [}"
+        tokenize_into_words = TokenizeIntoLowerWordsPrecalculation().tokenize_into_words
+        self.assertEqual(tokenize_into_words(normal), ['this', 'is', "a'n", 'example', '12', '356', 'str', 'ing', 'test'])
```
Comment (phSch08) [inline tests/precalculations/test_text_preprocessor.py]: line to long

File: shift_detector/precalculations/text_preprocessor.py [hunk 15]
```diff
@@ -0,0 +1,35 @@
+import re
```
Comment (phSch08) [inline shift_detector/precalculations/text_preprocessor.py]: Rename file to ...precalculation

File: tests/precalculations/test_text_preprocessor.py [hunk 16]
```diff
@@ -0,0 +1,73 @@
+import unittest
```
Comment (phSch08) [inline tests/precalculations/test_text_preprocessor.py]: rename file to ...precalculation

**Pull request merged**